rasa/core/policies/policy.py

```diff
@@ -293,6 +293,78 @@ def _default_predictions(domain: Domain) -> List[float]:
         return [0.0] * domain.num_actions
 
 
+class PolicyPrediction:
+    """Stores information about the prediction of a `Policy`."""
+
+    def __init__(
+        self,
+        probabilities: List[float],
+        policy_priority: int,
+        events: Optional[List[Event]] = None,
+        optional_events: Optional[List[Event]] = None,
+        is_end_to_end_prediction: bool = False,
+    ) -> None:
+        """Creates a `PolicyPrediction`.
+
+        Args:
+            probabilities: The probabilities for each action.
+            policy_priority: The priority of the policy which made the prediction.
+            events: Events which the `Policy` needs to have applied to the tracker
+                after the prediction. These events are applied independent of whether
+                the policy wins against other policies or not. Be careful which events
+                you return as they can potentially influence the conversation flow.
+            optional_events: Events which the `Policy` needs to have applied to the
+                tracker after the prediction in case it wins. These events are only
+                applied in case the policy's prediction wins. Be careful which events
+                you return as they can potentially influence the conversation flow.
+            is_end_to_end_prediction: `True` if the prediction used the text of the
+                user message instead of the intent.
+        """
+        self.probabilities = probabilities
+        self.policy_priority = (policy_priority,)
+        self.events = events or []
+        self.optional_events = optional_events or []
+        self.is_end_to_end_prediction = is_end_to_end_prediction
+
+    def __eq__(self, other: Any) -> bool:
+        """Checks if the two objects are equal.
+
+        Args:
+            other: Any other object.
+
+        Returns:
+            `True` if it's the same classes and the values are the same.
+        """
+        if not isinstance(other, PolicyPrediction):
+            return False
+
+        return (
+            self.probabilities == other.probabilities
+            and self.policy_priority == other.policy_priority
+            and self.events == other.events
+            and self.optional_events == other.events
+            and self.is_end_to_end_prediction == other.is_end_to_end_prediction
+        )
+
+    @property
+    def max_index(self) -> int:
```

Comment:
I don't think `max_index` is intuitive name, in numpy it is called `argmax`, or we can call it here explicitly `max_confidence_index` or smth